plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -417,11 +416,10 @@ public static DoubleWriteFunction oracleDoubleWriteFunction()
         return ((statement, index, value) -> ((OraclePreparedStatement) statement).setBinaryDouble(index, value));
     }
 
-    private SliceWriteFunction oracleCharWriteFunction(CharType charType)
+    private SliceWriteFunction oracleCharWriteFunction()
     {
-        return (statement, index, value) -> {
-            statement.setString(index, Chars.padSpaces(value, charType).toStringUtf8());
-        };
+        return (statement, index, value) ->
+                ((OraclePreparedStatement) statement).setFixedCHAR(index, value.toStringUtf8());
```

Comment:
Dumb question, what's the effective difference between `setFixedCHAR` and padding ourselves?

Reply:
Quoted from OraclePreparedStatement doc. setFixedCHAR will compare CHAR value without padding.
> Sets the disignated parameter to a String and executes a non-padded comparison with a SQL CHAR.

Not sure about how Oracle deal with char type containing unicode value in max length. So I tried to use setFixedCHAR and let oracle to handle it.

[Note]: I've checked Chars.padSpaces(value, charType).toStringUtf8(), it seems we do the right thing in the test case ->  Created a 2000 length string.

Reply:
Which test case did fail before the change here (and how)?

Reply:
after changing tests in `testCharUnicodeReadMapping`
```java
testTypeReadMapping("read_char_unicode",
                unicodeTests(charDataType(CHAR), codePoints(), MAX_CHAR_ON_READ),
```
to 
```java
SqlDataTypeTest.create()
                // the number of Unicode code points in 攻殻機動隊 is 5, and in 😂 is 1.
                .addRoundTrip("char(5 char)", "'攻殻機動隊'", createCharType(5), "CAST('攻殻機動隊' AS CHAR(5))")
                .addRoundTrip("char(13 char)", "'攻殻機動隊'", createCharType(13), "CAST('攻殻機動隊' AS CHAR(13))")
                .addRoundTrip(format("char(%d char)", MAX_CHAR_ON_READ), "'攻殻機動隊'",
                        createCharType(MAX_CHAR_ON_READ), format("CAST('攻殻機動隊' AS char(%d))", MAX_CHAR_ON_READ))
                .addRoundTrip("char(1 char)", "'\ud83d\ude02'", createCharType(1), "CAST('\ud83d\ude02' AS CHAR(1))")
                .addRoundTrip("char(6 char)", "'\ud83d\ude02'", createCharType(6), "CAST('\ud83d\ude02' AS CHAR(6))")
```
throws assertion exception.
```
java.lang.AssertionError: [Rows] 
Expecting:
  <>
to contain exactly in any order:
  <[(found)]>
but could not find the following elements:
  <(found)>
```
The error comes from this test case.
```java
.addRoundTrip(format("char(%d char)", MAX_CHAR_ON_READ), "'攻殻機動隊'", 
        createCharType(MAX_CHAR_ON_READ), format("CAST('攻殻機動隊' AS char(%d))", MAX_CHAR_ON_READ))
```

Reply:
Thanks for explanation.

Can you please add this single `SqlDataTypeTest` test case as part of the `Fix oracle incorrect char padding` commit, so that it stands as a documentation what case this is fixing? 

Then in `Convert oracle char unicode tests to SqlDataTypeTest` you will merge this one new test case  with all others. Thus the end result will be the same, but the fix commit will be better documented.